src/preview/VideoPreview.test.tsx

```diff
@@ -0,0 +1,190 @@
+		setup(<VideoPreview show={false} src={''} onClose={onClose} />);
+		expect(screen.queryByTestId('video')).not.toBeInTheDocument();
+	});
+
+	test('Additional data are visible', () => {
+		const onClose = jest.fn();
+		setup(
+			<VideoPreview
+				show
+				src={''}
+				onClose={onClose}
+				filename="video file name"
+				extension="mp4"
+				size="18MB"
+			/>
+		);
+		expect(screen.getByText(/video file name/i)).toBeVisible();
+		expect(screen.getByText(/mp4/i)).toBeVisible();
+		expect(screen.getByText(/18MB/i)).toBeVisible();
+	});
+
+	test('Escape key close the preview', async () => {
+		const onClose = jest.fn();
+		const { user } = setup(<VideoPreview show src={''} onClose={onClose} />);
+		await user.keyboard(KEYBOARD_KEY.ESC);
+		expect(onClose).toHaveBeenCalled();
+	});
+
+	test('Click on actions calls onClose if event is not stopped by the action itself', async () => {
+		const onClose = jest.fn<void, Parameters<VideoPreviewProps['onClose']>>((ev) => {
+			ev.preventDefault();
+		});
+		const actions: VideoPreviewProps['actions'] = [
+			{
+				id: 'action1',
+				icon: 'Activity',
+				onClick: jest.fn()
+			},
+			{
+				id: 'action2',
+				icon: 'People',
+				onClick: jest.fn((ev: React.MouseEvent<HTMLButtonElement> | KeyboardEvent) => {
+					ev.preventDefault();
+				}),
+				disabled: true
+			}
+		];
+
+		const closeAction: VideoPreviewProps['closeAction'] = {
+			id: 'closeAction',
+			icon: 'Close'
+		};
+		const { user } = setup(
+			<VideoPreview
+				show
+				src={''}
+				onClose={onClose}
+				actions={actions}
+				filename="video name"
+				closeAction={closeAction}
+			/>
+		);
+		const action1Item = screen.getByRoleWithIcon('button', { icon: 'icon: Activity' });
+		const action2Item = screen.getByRoleWithIcon('button', { icon: 'icon: People' });
+		const closeActionItem = screen.getByRoleWithIcon('button', { icon: 'icon: Close' });
+		expect(action1Item).toBeVisible();
+		expect(action2Item).toBeVisible();
+		expect(action2Item).toBeDisabled();
+		expect(closeActionItem).toBeVisible();
+		// click on action 1 is propagated and calls onClose
+		await user.click(action1Item);
+		expect(actions[0].onClick).toHaveBeenCalled();
+		expect(onClose).toHaveBeenCalledTimes(1);
+		// click on action 2 skips the handler of the action since it is disabled and calls onClose
+		await user.click(action2Item);
+		expect(actions[1].onClick).not.toHaveBeenCalled();
+		expect(onClose).toHaveBeenCalledTimes(1);
+		// click on close action is the onClose itself
+		await user.click(closeActionItem);
+		expect(onClose).toHaveBeenCalledTimes(2);
+		// click on filename is equivalent to a click on the overlay, so onClose is called
+		await user.click(screen.getByText(/video name/i));
+		expect(onClose).toHaveBeenCalledTimes(3);
+	});
+
+	it('should not render the video when canPlayType return empty string on mime type(mime type not supported)', () => {
+		jest.spyOn(utils, 'videoCanPlayType').mockReturnValue('');
```

Comment:
There is no need to have the util, we can spyOn the video prototype function directly
```suggestion
		jest.spyOn(HTMLVideoElement.prototype, 'canPlayType').mockReturnValue('');
```